src/main/java/org/reaktivity/nukleus/oauth/internal/OAuthRealms.java

```diff
@@ -167,4 +218,43 @@ public JsonWebKey supplyKey(
 
         return keysByKid;
     }
+
+    private final class OAuthRealmObject
+    {
+        private static final int MAX_SCOPES = 48;
+
+        private final Map<String, Long> scopeBitsByName = new CopyOnWriteHashMap<>();
+
+        private long realmBit;
+        private long nextScopeBitShift;
+
+        private OAuthRealmObject(long realmBit)
+        {
+            this.realmBit = realmBit;
+        }
+
+        private boolean scopeBitAssigned(
+            String scope)
+        {
+            return scopeBitsByName.containsKey(scope);
+        }
+
+        private boolean addScopeBit(
+            String scope)
+        {
+            final long nextScopeBit = scopeBitsByName.size() < MAX_SCOPES ? (1L << nextScopeBitShift++) : -1;
+            if(nextScopeBit < 0)
+            {
+                return false;
+            }
+            scopeBitsByName.put(scope, nextScopeBit);
+            return true;
```

Comment:
Please rename to `supplyScopeBit(...)` and make use of built-in method `computeIfAbsent`, e.g.
```
return scopeBitsByName.size() < MAX_SCOPES &&
       scopeBitsByName.computeIfAbsent(scope, s -> 1L << nextScopeBitShift++);
```